### PR TITLE
Fix 26.6 mul/div operators 

### PR DESCRIPTION
**Pull request by dfrg**
Actually perform operations in 26.6 Also reworks scaling a bit in skrifa to account for this. Checked fauntlet against the Noto library to confirm this doesn't break anything.

Supersedes #1698

### CODE REVIEW

File: font-types/src/fixed.rs [hunk 1]
```diff
@@ -137,96 +137,120 @@ macro_rules! fixed_impl {
...
+        } else {
+            q as i32
+        })
+    }
+}
+
+impl Mul for F26Dot6 {
+    type Output = Self;
+
+    #[allow(clippy::suspicious_arithmetic_impl)]
```
Comment (cmyr) [inline font-types/src/fixed.rs]: I'd suggest using `#[expect]` instead of `#[allow]` in basically all cases now? it has the nice advantage of letting us know if the lint has gone away or a future revision makes it unnecessary.

File: font-types/src/fixed.rs [hunk 2]
```diff
@@ -523,4 +549,24 @@ mod tests {
...
+        );
+        assert_eq!(
+            F26Dot6::from_f64(0.5) / F26Dot6::from_f64(2.0),
+            F26Dot6::from_f64(0.25)
+        );
+        assert_eq!(
+            F26Dot6::from_f64(0.5) / F26Dot6::from_f64(-2.4),
+            F26Dot6::from_f64(-0.20833333333333334)
+        );
+    }
```
Comment (cmyr) [inline font-types/src/fixed.rs:588]: might as well also include the test from the original PR?

https://github.com/daltonmaag/fontations/blob/e08bf25a03b5851e0ae79fa0dc98c6f6afb56ae1/font-types/src/fixed.rs#L515

**Review by cmyr — APPROVED** [review]
thanks, will be nice to get this tided up.

File: font-types/src/fixed.rs (hunk 1 above)
Comment (dfrg) [reply font-types/src/fixed.rs]: Makes sense, will do in the future. I ended up changing the code so the semantics match Fixed and that removed the need for the clippy lints.

File: font-types/src/fixed.rs (hunk 2 above)
Comment (dfrg) [reply font-types/src/fixed.rs:588]: Copied the identity tests from that PR and added a few more to capture some edge cases.
